generate-list-objects-filter-extensions.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/sh
+
+if [ $# -gt 0 ]; then
+
+	# ARGS has one argument per line
+	ARGS=`echo $@ | xargs printf '%s\n'`
+
+	# INVALID_ARGS is those ARGS that don't look like [path/]library.a
+	INVALID_ARGS=`echo "$ARGS" | grep -v '\b[A-Za-z0-9_]\+\.a$'`
```

Comment:
Disallowed simply because it makes linking more complicated: foo.a must define a symbol called `filter_extension_foo`.

Updated the script to fail differently for two different errors - ie, not a path to a library, and name of library is not a valid C symbol.